src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/DoNotUseOutAttributeStringPInvokeParameters.cs

```diff
@@ -0,0 +1,78 @@
+    public sealed class DoNotUseOutAttributeStringPinvokeParametersAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1417";
+
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.DoNotUseOutAttributeStringPInvokeParametersTitle), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.DoNotUseOutAttributeStringPInvokeParametersMessage), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.DoNotUseOutAttributeStringPInvokeParametersDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        internal static DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(
+                                                        RuleId,
+                                                        s_localizableTitle,
+                                                        s_localizableMessage,
+                                                        DiagnosticCategory.Interoperability,
+                                                        RuleLevel.BuildWarning,
+                                                        description: s_localizableDescription,
+                                                        isPortedFxCopRule: false,
+                                                        isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(
+                compilationContext =>
+                {
+                    INamedTypeSymbol? outAttributeType = compilationContext.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesOutAttribute);
+                    if (outAttributeType == null)
+                    {
+                        return;
+                    }
+
+                    compilationContext.RegisterSymbolAction(symbolContext => AnalyzeSymbol(symbolContext, outAttributeType), SymbolKind.Method);
+                });
+        }
+
+        private static void AnalyzeSymbol(SymbolAnalysisContext context, INamedTypeSymbol outAttributeType)
+        {
+            var methodSymbol = (IMethodSymbol)context.Symbol;
+            if (methodSymbol == null)
```

Comment:
This null check is redundant.

Reply:
@mavasani I don't see how it is redundant. Is there a precondition somewhere?

Reply:
This will always be a non-null symbol for the analyzer callback. That is a contract guaranteed by the analyzer driver.